### PR TITLE
Update example to show how to do navigation with tabbed pages

### LINKED ISSUE #52: [Update] 08 Tabbed Navigation
### Sample

08-TabbedNavigation

#### Reason for Update

This sample is currently stubbed and needs actual implementation.

TODO

- Create tabs via querystring
- Select a tab (not the first one) via the querystring
- Show how to change tabs with the NavigationService

### PR DESCRIPTION
**Pull request by ScarlettCode**
closes #52

This example shows the following

Creating a tab page in xaml and navigating to a tab with the query string
Create tabs at runtime with the query string
Navigating to a tab via the querystring
Changing to another tab with the navigation service

### CODE REVIEW

File: 08-TabbedNavigation/src/PrismSample/ViewModels/MainPageViewModel.cs [hunk 1]
```diff
@@ -1,11 +1,37 @@
...
 using Prism.Mvvm;
+using Prism.Navigation;
+using PrismSample.Views;
+using Xamarin.Forms;
 
 namespace PrismSample.ViewModels
 {
     public class MainPageViewModel : BindableBase
     {
+        private readonly INavigationService _navigationService;
```
Comment (dansiegel) [inline 08-TabbedNavigation/src/PrismSample/ViewModels/MainPageViewModel.cs]: readonly property/field declarations shouldn't be mixed... keep it consistent here `private INavigationService _navigationService { get; }`

**Comment by ScarlettCode** [comment]
- Should we not be showing `IActiveAware` as `INavigationAware` only gets called for the first tab you land on which isn't super helpful compared to IActiveAware? Or be showing both?

- TabCPageViewModel uses the SelectTabAsync to set a tab. I need to correct the asyncness of that but I think that's what you referring to?

File: 08-TabbedNavigation/src/PrismSample/Views/TabsPage.xaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+﻿<?xml version="1.0" encoding="utf-8" ?>
+<TabbedPage xmlns="http://xamarin.com/schemas/2014/forms"
+            xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
+            xmlns:prism="http://prismlibrary.com"
+            xmlns:views="clr-namespace:PrismSample.Views;assembly=PrismSample"
+            prism:ViewModelLocator.AutowireViewModel="True"
```
Comment (dansiegel) [inline 08-TabbedNavigation/src/PrismSample/Views/TabsPage.xaml]: Do not include the ViewModelLocator.AutowireViewModel... it's not needed for any page in this sample!

File: 08-TabbedNavigation/src/PrismSample/Views/TabsPage.xaml [hunk 3]
```diff
@@ -0,0 +1,15 @@
+﻿<?xml version="1.0" encoding="utf-8" ?>
+<TabbedPage xmlns="http://xamarin.com/schemas/2014/forms"
```
Comment (dansiegel) [inline 08-TabbedNavigation/src/PrismSample/Views/TabsPage.xaml]: Be sure to update the Application Resources style for TabbedPage to include the following: `ApplyToDerivedTypes="True"`

File: 08-TabbedNavigation/src/PrismSample/Views/TabCPage.xaml [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
+             xmlns:prism="http://prismlibrary.com"
+             prism:ViewModelLocator.AutowireViewModel="True"
+             x:Class="PrismSample.Views.TabCPage"
+             Title="Tab C">
+
+  <StackLayout>
+    <Label Text="Tab C" HorizontalOptions="CenterAndExpand"></Label>
+    <Button Text="Go To Tab A" HorizontalOptions="CenterAndExpand" Command="{Binding TabACommand}"></Button>
+  </StackLayout>
```
Comment (dansiegel) [inline 08-TabbedNavigation/src/PrismSample/Views/TabCPage.xaml]: - Center the StackLayout Vertically
- The label should be using `HorizontalTextAlignment="Center"` not the HorizontalOptions
- All elements like Label / Button that you are not setting complex properties on should be self closing `<Label />`

File: 08-TabbedNavigation/src/PrismSample/Views/TabCPage.xaml [hunk 5]
```diff
@@ -0,0 +1,14 @@
+﻿<?xml version="1.0" encoding="utf-8" ?>
+<ContentPage xmlns="http://xamarin.com/schemas/2014/forms"
+             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
+             xmlns:prism="http://prismlibrary.com"
+             prism:ViewModelLocator.AutowireViewModel="True"
```
Comment (dansiegel) [inline 08-TabbedNavigation/src/PrismSample/Views/TabCPage.xaml]: Remove the ViewModelLocator.AutowireViewModel property as it is not needed here.

File: 08-TabbedNavigation/src/PrismSample/Views/TabBPage.xaml [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+             xmlns:prism="http://prismlibrary.com"
+             prism:ViewModelLocator.AutowireViewModel="True"
+             x:Class="PrismSample.Views.TabBPage"
+             Title="Tab B">
+  
+  <StackLayout>
+    <Label Text="Tab B" HorizontalOptions="CenterAndExpand"></Label>
+  </StackLayout>
+
+</ContentPage>
```
Comment (dansiegel) [inline 08-TabbedNavigation/src/PrismSample/Views/TabBPage.xaml]: See comments on TabC

File: 08-TabbedNavigation/src/PrismSample/Views/TabAPage.xaml [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+             xmlns:prism="http://prismlibrary.com"
+             prism:ViewModelLocator.AutowireViewModel="True"
+             x:Class="PrismSample.Views.TabAPage"
+             Title="Tab A">
+  
+  <StackLayout>
+    <Label Text="Tab A" HorizontalOptions="CenterAndExpand"></Label>
+  </StackLayout>
+
+</ContentPage>
```
Comment (dansiegel) [inline 08-TabbedNavigation/src/PrismSample/Views/TabAPage.xaml]: See comments on TabC

File: 08-TabbedNavigation/src/PrismSample/Views/MainPage.xaml [hunk 8]
```diff
@@ -7,10 +7,11 @@
...
 
-    <StackLayout>
-        <Label Text="Welcome to Xamarin.Forms!" 
-           HorizontalOptions="Center"
-           VerticalOptions="CenterAndExpand" />
-    </StackLayout>
+  <StackLayout VerticalOptions="CenterAndExpand">
+
+    <Button Text="Create Tabs At Runtime" Command="{Binding RuntimeTabsAction}"></Button>
+    <Button Text="Go To Tab B" Command="{Binding TabBCommand}"></Button>
```
Comment (dansiegel) [inline 08-TabbedNavigation/src/PrismSample/Views/MainPage.xaml]: You have an invalid binding... also `<Button />`

File: 08-TabbedNavigation/src/PrismSample/ViewModels/TabCPageViewModel.cs [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+using Prism.Navigation;
+using Prism.Navigation.TabbedPages;
+using PrismSample.Views;
+
+namespace PrismSample.ViewModels
+{
+    public class TabCPageViewModel : BindableBase
+    {
+        private readonly INavigationService _navigationService;
+        public DelegateCommand TabACommand { get; set; }
```
Comment (dansiegel) [inline 08-TabbedNavigation/src/PrismSample/ViewModels/TabCPageViewModel.cs]: see comments on MainPageViewModel

File: 08-TabbedNavigation/src/PrismSample/ViewModels/MainPageViewModel.cs [hunk 10]
```diff
@@ -1,11 +1,41 @@
...
+            TabBCommand = new DelegateCommand(TabBAction);
+            RuntimeTabsCommand = new DelegateCommand(RuntimeTabsAction);
+        }
+
+        private async void RuntimeTabsAction()
+        {
+            await _navigationService.NavigateAsync("TabbedPage" +
+                                             $"?{KnownNavigationParameters.CreateTab}=TabCPage" +
+                                             $"&{KnownNavigationParameters.CreateTab}=TabBPage" +
+                                             $"&{KnownNavigationParameters.CreateTab}=TabAPage");
```
Comment (dansiegel) [inline 08-TabbedNavigation/src/PrismSample/ViewModels/MainPageViewModel.cs]: // KnowNavigationParameters.CreateTab == createTab
`createTab=TabCPage&createTab=TabBPage`

File: 08-TabbedNavigation/src/PrismSample/ViewModels/MainPageViewModel.cs [hunk 11]
```diff
@@ -1,11 +1,41 @@
...
+        }
+
+        private async void TabBAction()
+        {
+            var p = new NavigationParameters();
+            p.Add("ActiveCount", 0);
+
+            await _navigationService.NavigateAsync("TabsPage?selectedTab=TabBPage", p);
```
Comment (dansiegel) [inline 08-TabbedNavigation/src/PrismSample/ViewModels/MainPageViewModel.cs]: as a comment:

// await _navigationService.NavigateAsync("TabsPage?selectedTab=TabBPage", ("ActiveCount", 0));

File: 08-TabbedNavigation/src/PrismSample/ViewModels/TabsPageViewModel.cs [hunk 12]
```diff
@@ -0,0 +1,16 @@
...
+using System.Linq;
+
+namespace PrismSample.ViewModels
+{
+    public class TabsPageViewModel : BindableBase
+    {
+        public TabsPageViewModel()
+        {
+
+        }
```
Comment (dansiegel) [inline 08-TabbedNavigation/src/PrismSample/ViewModels/TabsPageViewModel.cs]: Let's add a Title property... and be sure to wrap this page in a NavigationPage

File: 08-TabbedNavigation/src/PrismSample/Views/TabsPage.xaml [hunk 13]
```diff
@@ -0,0 +1,14 @@
...
+            xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
+            xmlns:prism="http://prismlibrary.com"
+            xmlns:views="clr-namespace:PrismSample.Views;assembly=PrismSample"
+            x:Class="PrismSample.Views.TabsPage">
```
Comment (dansiegel) [inline 08-TabbedNavigation/src/PrismSample/Views/TabsPage.xaml]: We'll want to bind a the Title to the ViewModel

**Comment by Kylar182** [comment]
Considering t here's nothing at all in the master branch can we have this to just have something? Even if it's not perfect?

**Comment by brianlagunas** [comment]
Closing as requested changes have not been made in a timely manner

**Pull request closed without merging**